pkg/scheduler/core/extender.go

```diff
@@ -299,8 +301,14 @@ func (h *HTTPExtender) Filter(
 
 	if h.nodeCacheCapable && result.NodeNames != nil {
 		nodeResult = make([]*v1.Node, 0, len(*result.NodeNames))
-		for i := range *result.NodeNames {
-			nodeResult = append(nodeResult, nodeNameToInfo[(*result.NodeNames)[i]].Node())
+		for _, nodeName := range *result.NodeNames {
+			if node, ok := nodeNameToInfo[nodeName]; ok {
+				nodeResult = append(nodeResult, node.Node())
+			} else {
+				return nil, nil, fmt.Errorf(
+					"extender: %v claims a filtered node: %v but the node is not found in nodeNameToInfo map",
```

Comment:
```suggestion
					"extender %q claims a filtered node %q which is not found in nodeNameToInfo map",
```

Reply:
This is to align with preemption error message, do I also need to make simliar change for preemption code?  Such as:

```
		return nil, fmt.Errorf("extender: %v claims to preempt pod (UID: %v) on node: %v, but the pod is not found on that node",
			h.extenderURL, metaPod, nodeInfo.Node().Name)
	}

	return nil, fmt.Errorf("extender: %v claims to preempt on node: %v but the node is not found in nodeNameToInfo map",
```

Reply:
That's fine. Just a nit.